internal/codespaces/rpc/invoker.go

```diff
@@ -239,6 +239,7 @@ func (i *invoker) StartSSHServerWithOptions(ctx context.Context, options StartSS
 }
 
 func listenTCP() (*net.TCPListener, error) {
+	// We will end up using this same address to connect, so specify the IP also or the connect will fail
```

Comment:
I'm not following why 127.0.0.1 works in this scenario but not the other scenario? Is it because we are asking for a random port with `:0`? Could we not just bind to `:0` similar to `:<port>` in the case where we want a specific port number?

Reply:
Specifically for the invoker, the CLI will also connect to the listener using `listener.Addr().String()` (see line 77 in this same file) and if we just specific `:0` then `listener.Addr().String()` will be something like `[::]:<port>` which is not an address you can actually use to connect to.

I had considered using `:0` here also and just parsing out the selected port and building the connect address separately, but that just felt a little convoluted when we could just keep it like this.